src/kakarot/library.cairo

```diff
@@ -331,6 +211,28 @@ namespace Kakarot {
         return (starknet_contract_address=starknet_contract_address);
     }
 
+    // @notice Get the ExecutionContext address from the transaction
+    // @dev When to=0, it's a deploy tx so we first compute the target address
+    // @param to The transaction to parameter
+    // @param origin The transaction origin parameter
+    // @return the target evm address
+    func resolve_to{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(to: felt, origin: felt) -> felt {
+        alloc_locals;
+        if (to != 0) {
+            return to;
+        }
+        // TODO: read the nonce from the provided origin address, otherwise in view mode this will
```

Comment:
afaik we cannot get the nonce of the provided origin address if it is an eoa (and it *must* be an eoa)

Reply:
we asked for this few weeks ago but it's not possible yet. We miss a `get_nonce` syscall, and it's the same in cairo1 I guess